[BUG] Cannot find module '@zkochan/js-yaml' after package updates

### Describe the bug

For about 2 weeks now (possibly earlier but I wasn't package updating then) when a package update is done (via updating the .projenrc.ts file) for the aws-prototyping-sdk package (i.e. 0.17.2 -> 0.17.4, but also prior package upgrades) a package will fail to be found when doing a build after the `yarn projen` command has updated the packages.



### Expected Behavior

Not to get an error when doing a fresh build after upgrading this package.

### Current Behavior

When invoking an nx build command after upgrading the package I'll get the below error:


 I'll get the error: 
```bash
 >  NX   Cannot find module '@zkochan/js-yaml'

   Require stack:
   - /Users/joshtoth/repo/node_modules/nx/src/plugins/js/lock-file/utils/pnpm-normalizer.js
   - /Users/joshtoth/repo/node_modules/nx/src/plugins/js/lock-file/pnpm-parser.js
   - /Users/joshtoth/repo/node_modules/nx/src/plugins/js/lock-file/lock-file.js
   - /Users/joshtoth/repo/node_modules/nx/src/plugins/js/index.js
   - /Users/joshtoth/repo/node_modules/nx/src/utils/nx-plugin.js
   - /Users/joshtoth/repo/node_modules/nx/src/project-graph/build-nodes/workspace-projects.js
   - /Users/joshtoth/repo/node_modules/nx/src/project-graph/build-nodes/index.js
   - /Users/joshtoth/repo/node_modules/nx/src/project-graph/build-project-graph.js
   - /Users/joshtoth/repo/node_modules/nx/src/project-graph/project-graph.js
   - /Users/joshtoth/repo/node_modules/nx/src/utils/project-graph-utils.js
   - /Users/joshtoth/repo/node_modules/nx/src/tasks-runner/utils.js
   - /Users/joshtoth/repo/node_modules/nx/src/tasks-runner/run-command.js
   - /Users/joshtoth/repo/node_modules/nx/src/command-line/run-many.js
   - /Users/joshtoth/repo/node_modules/nx/src/command-line/nx-commands.js
   - /Users/joshtoth/repo/node_modules/nx/bin/init-local.js
   - /Users/joshtoth/repo/node_modules/nx/bin/nx.js
```

### Reproduction Steps

With a repository using `aws-prototyping-sdk` packages, upgrade the package to a newer version: 0.17.2 -> 0.17.4 and then run `yarn projen && npx nx run-many --target=build --all` and the build will fail.

it's specifically the nx build command that fails and targeting a specific package will fail regardless of the package.

`npx nx run website:build`
`npx nx run infra:build`

etc.

This also occurs when _downgrading_ packages.

### Possible Solution

_No response_

### Additional Information/Context

## Current workaround:

Delete the yarn lockfile at the root of the repository, then run `yarn && yarn projen && npx nx run-many --target=build --all`

## Machine info: 

I am currently using an M1 mac running 13.3.1. Although I have seen the same issue replicated on an intel mac as well.

I have also seen this on another machine, although I don't know it's specs.

I thought perhaps it's just my setup but I've seen this now in 3 places so raising this ticket. Not sure if this is something that can be fixed on this packages end, but the workaround is good for visibility.

### PDK version used

0.17.2, 0.17.4

### What languages are you seeing this issue on?

_No response_

### Environment details (OS name and version, etc.)

macOS 13.3.1

--- a/package.json
+++ b/package.json
@@ -71,6 +71,8 @@
   "pnpm": {
     "overrides": {
       "@types/babel__traverse": "7.18.2",
+      "wrap-ansi": "^7.0.0",
+      "@zkochan/js-yaml": "npm:js-yaml@4.1.0",
       "@types/prettier": "2.6.0",
       "ansi-regex": "^5.0.1",
       "underscore": "^1.12.1",

--- a/packages/nx-monorepo/src/nx-monorepo.ts
+++ b/packages/nx-monorepo/src/nx-monorepo.ts
@@ -379,7 +379,11 @@ export class NxMonorepoProject extends TypeScriptProject {
 
     this.addPeerDeps("nx@^16");
     this.addDeps("aws-cdk-lib", "constructs", "cdk-nag"); // Needed as this can be bundled in aws-prototyping-sdk
-    this.package.addPackageResolutions("@types/babel__traverse@7.18.2");
+    this.package.addPackageResolutions(
+      "@types/babel__traverse@7.18.2",
+      "wrap-ansi@^7.0.0",
+      "@zkochan/js-yaml@npm:js-yaml@4.1.0"
+    );
 
     if (options.monorepoUpgradeDeps !== false) {
       this.addDevDeps("npm-check-updates", "syncpack");

